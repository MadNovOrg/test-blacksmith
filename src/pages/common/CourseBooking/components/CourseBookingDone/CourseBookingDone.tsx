import { Alert, Box, Container, Link, Stack, Typography } from '@mui/material'
import React, { useEffect, useMemo } from 'react'
import { Trans, useTranslation } from 'react-i18next'
import { useSearchParams } from 'react-router-dom'
import { useQuery } from 'urql'

import { StepsNavigation } from '@app/components/StepsNavigation'
import { Sticky } from '@app/components/Sticky'
import { useAuth } from '@app/context/auth'
import {
  GetOrderForBookingDoneQuery,
  GetOrderForBookingDoneQueryVariables,
  Payment_Methods_Enum,
} from '@app/generated/graphql'
import { useFetcher } from '@app/hooks/use-fetcher'
import { ORDER_FOR_BOOKING_DONE } from '@app/queries/order/get-order'
import {
  MUTATION as DELETE_TEMP_PROFILE,
  ResponseType as DeleteTempProfileResponseType,
  ParamsType as DeleteTempProfileParamsType,
} from '@app/queries/profile/delete-temp-profile'

const completedSteps = ['details', 'review', 'payment']

export const CourseBookingDone: React.FC<
  React.PropsWithChildren<unknown>
> = () => {
  const { t } = useTranslation()
  const { profile, acl } = useAuth()
  const fetcher = useFetcher()
  const [searchParams] = useSearchParams()
  const orderId = searchParams.get('order_id') as string

  useEffect(() => {
    if (!profile) return

    fetcher<DeleteTempProfileResponseType, DeleteTempProfileParamsType>(
      DELETE_TEMP_PROFILE,
      { email: profile.email }
    )
  }, [fetcher, profile])

  const [{ data, error }] = useQuery<
    GetOrderForBookingDoneQuery,
    GetOrderForBookingDoneQueryVariables
  >({
    query: ORDER_FOR_BOOKING_DONE,
    variables: { orderId },
    pause: !orderId,
  })
  const order = data?.order
  const noOrder = order === null || error

  const steps = useMemo(() => {
    if (!order) return []

    return [
      {
        key: 'details',
        label: t('pages.book-course.step-1'),
      },
      {
        key: 'review',
        label: t('pages.book-course.step-2'),
      },
      order.paymentMethod === Payment_Methods_Enum.Cc
        ? { key: 'payment', label: t('pages.book-course.step-3') }
        : null,
    ].filter(Boolean)
  }, [t, order])

  return (
    <Box bgcolor="grey.100" height="100%">
      <Container maxWidth="lg" sx={{ py: 3 }}>
        <Box flex={1} display="flex">
          <Box width={300} display="flex" flexDirection="column" pr={4}>
            <Sticky top={20}>
              {steps.length ? (
                <Box mb={7}>
                  <Typography variant="h2" mb={2}>
                    {t('pages.book-course.title-complete')}
                  </Typography>
                </Box>
              ) : null}

              <StepsNavigation
                completedSteps={completedSteps}
                currentStepKey={null}
                steps={steps}
                data-testid="create-course-nav"
              />
            </Sticky>
          </Box>

          <Box flex={1}>
            <Box>
              <Stack flex={1} alignItems="center" textAlign="center">
                {noOrder ? (
                  <Alert color="error" severity="error" sx={{ mb: 4 }}>
                    {t('errors.loading-order')}
                  </Alert>
                ) : (
                  <Alert variant="outlined" color="success" sx={{ mb: 4 }}>
                    <Trans
                      i18nKey="pages.book-course.order-success-info"
                      values={{ orderNumber: order?.xeroInvoiceNumber }}
                    >
                      {acl.canViewOrders() ? (
                        <Link href={`/orders/${order?.id}`} />
                      ) : (
                        <></>
                      )}
                    </Trans>
                  </Alert>
                )}

                {noOrder ? null : (
                  <Typography variant="subtitle1">
                    {t('pages.book-course.order-success-msg')}
                  </Typography>
                )}
              </Stack>
            </Box>
          </Box>
        </Box>
      </Container>
    </Box>
  )
}
