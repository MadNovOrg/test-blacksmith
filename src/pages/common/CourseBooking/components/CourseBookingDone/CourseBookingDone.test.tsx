import React from 'react'
import { MemoryRouter, Routes, Route } from 'react-router-dom'

import { render } from '@test/index'

import { CourseBookingDone } from './CourseBookingDone'

describe('CourseBookingDone', () => {
  it('matches snapshot', async () => {
    const view = render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path="/" element={<CourseBookingDone />} />
        </Routes>
      </MemoryRouter>
    )

    expect(view).toMatchSnapshot()
  })
})
