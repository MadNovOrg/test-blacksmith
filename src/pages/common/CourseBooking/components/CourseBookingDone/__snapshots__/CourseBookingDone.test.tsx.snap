// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseBookingDone matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-orciji"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-nd3jgp-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-i3jcei"
          >
            <div
              class="MuiBox-root css-1n9f69o"
            >
              <div
                class="MuiBox-root css-1swh5hj"
              >
                <div
                  class="MuiBox-root css-1r1aq4"
                >
                  <h2
                    class="MuiTypography-root MuiTypography-h2 css-17li6z0-MuiTypography-root"
                  >
                    Book a course
                  </h2>
                </div>
                <div
                  class="MuiBox-root css-0"
                  data-testid="create-course-nav"
                >
                  <div
                    class="MuiBox-root css-ndrivx"
                    data-testid="step-item-1"
                  >
                    <span
                      class="css-1jr2wd7"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                        />
                      </svg>
                    </span>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-7iwe82-MuiTypography-root"
                    >
                      Booking details
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-ndrivx"
                    data-testid="step-item-2"
                  >
                    <span
                      class="css-cr7of8"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                        />
                      </svg>
                    </span>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-7iwe82-MuiTypography-root"
                    >
                      Review & confirm
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="css-lyd85o-MuiStack-root"
                >
                  <div
                    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedSuccess MuiAlert-outlined css-fs5m4j-MuiPaper-root-MuiAlert-root"
                    role="alert"
                  >
                    <div
                      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="SuccessOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiAlert-message css-acap47-MuiAlert-message"
                    >
                      Your order has been successfully submitted.
                    </div>
                  </div>
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle1 css-12kg9eg-MuiTypography-root"
                  >
                    Each attendee will receive an email with the necessary course instructions once this order has been processed and approved.
                  </h6>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-orciji"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-nd3jgp-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-i3jcei"
        >
          <div
            class="MuiBox-root css-1n9f69o"
          >
            <div
              class="MuiBox-root css-1swh5hj"
            >
              <div
                class="MuiBox-root css-1r1aq4"
              >
                <h2
                  class="MuiTypography-root MuiTypography-h2 css-17li6z0-MuiTypography-root"
                >
                  Book a course
                </h2>
              </div>
              <div
                class="MuiBox-root css-0"
                data-testid="create-course-nav"
              >
                <div
                  class="MuiBox-root css-ndrivx"
                  data-testid="step-item-1"
                >
                  <span
                    class="css-1jr2wd7"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                      />
                    </svg>
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-7iwe82-MuiTypography-root"
                  >
                    Booking details
                  </p>
                </div>
                <div
                  class="MuiBox-root css-ndrivx"
                  data-testid="step-item-2"
                >
                  <span
                    class="css-cr7of8"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                      />
                    </svg>
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-7iwe82-MuiTypography-root"
                  >
                    Review & confirm
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="css-lyd85o-MuiStack-root"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedSuccess MuiAlert-outlined css-fs5m4j-MuiPaper-root-MuiAlert-root"
                  role="alert"
                >
                  <div
                    class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                      data-testid="SuccessOutlinedIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiAlert-message css-acap47-MuiAlert-message"
                  >
                    Your order has been successfully submitted.
                  </div>
                </div>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 css-12kg9eg-MuiTypography-root"
                >
                  Each attendee will receive an email with the necessary course instructions once this order has been processed and approved.
                </h6>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "context": Object {
    "auth": Object {
      "activeRole": "user",
      "allowedRoles": Set {
        "user",
      },
      "changeRole": [MockFunction],
      "defaultRole": "user",
      "getJWT": [MockFunction],
      "loadProfile": [MockFunction],
      "loading": false,
      "login": [MockFunction],
      "logout": [MockFunction],
      "profile": Object {
        "addresses": Array [],
        "attributes": Array [],
        "avatar": "",
        "contactDetails": Array [],
        "createdAt": "",
        "dietaryRestrictions": null,
        "disabilities": null,
        "dob": "2002-04-06",
        "email": "john.smith@example.com",
        "familyName": "Smith",
        "fullName": "John Smith",
        "givenName": "John",
        "id": "cacb559d-b85d-5e64-b623-37252520ebda",
        "jobTitle": "Developer",
        "lastActivity": 2020-01-01T00:00:00.000Z,
        "organizations": Array [],
        "phone": "+44 1234555",
        "preferences": Array [],
        "roles": Array [
          Object {
            "role": Object {
              "id": "1",
              "name": "user",
            },
          },
        ],
        "tags": Array [],
        "title": "Mr",
        "trainer_role_types": Array [
          Object {
            "trainer_role_type": Object {
              "id": "1",
              "name": "principal",
            },
          },
        ],
        "updatedAt": "",
      },
      "reloadCurrentProfile": [MockFunction],
    },
  },
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
